File: cinema-manager/backend/src/main/java/com/example/cinema/managing/system/service/ContactService.java
package com.example.cinema.managing.system.service;

import com.example.cinema.managing.system.model.Contact;
import com.example.cinema.managing.system.repository.ContactRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import java.time.LocalDateTime;
import java.util.List;

@Service
public class ContactService {

    private static final Logger logger = LoggerFactory.getLogger(ContactService.class);

    @Autowired
    private ContactRepository contactRepository;

    public Contact createContact(Contact contact) {
        logger.info("Creating contact: {}", contact.getName());
        contact.setStatus("NEW");
        contact.setCreatedAt(LocalDateTime.now());
        contact.setUpdatedAt(LocalDateTime.now());
        Contact saved = contactRepository.save(contact);
        logger.info("Contact saved successfully with ID: {}", saved.getId());
        return saved;
    }

    public List<Contact> getAllContacts() {
        return contactRepository.findAllByOrderByCreatedAtDesc();
    }

    public List<Contact> getContactsByStatus(String status) {
        return contactRepository.findByStatusOrderByCreatedAtDesc(status);
    }

    public Contact updateContactStatus(String id, String status) {
        Contact contact = contactRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Contact not found"));
        contact.setStatus(status);
        contact.setUpdatedAt(LocalDateTime.now());
        return contactRepository.save(contact);
    }

    public void deleteContact(String id) {
        contactRepository.deleteById(id);
    }
}

File: cinema-manager/frontend/src/services/contactService.js
import API from './api';

const contactService = {
  submitContact: async (contactData) => {
    const response = await API.post('/contacts', contactData);
    return response.data;
  },

  getAllContacts: async () => {
    const response = await API.get('/contacts');
    return response.data;
  },

  getContactsByStatus: async (status) => {
    const response = await API.get(`/contacts/status/${status}`);
    return response.data;
  },

  updateContactStatus: async (id, status) => {
    const response = await API.put(`/contacts/${id}/status?status=${status}`);
    return response.data;
  },

  deleteContact: async (id) => {
    await API.delete(`/contacts/${id}`);
  }
};

export default contactService;
